addons/website/models/website_design.py

```diff
@@ -0,0 +1,44 @@
+from odoo import models, fields
+
+# Details but theme_id should not be blacklisted here.
+NOT_DESIGN_FIELDS = ['id', 'display_name', 'create_uid', 'create_date', 'write_uid', 'write_date', 'theme_id']
+
+class WebsiteDesign(models.Model):
+    _name = 'website.design'
+    _description = 'Website Design'
+
+    # DESIGN VARIABLES
+    layout = fields.Char(string='Layout', default="'full'")
```

Comment:
Maybe `layout` was not the right one to start with. For this kinda of variable, I actually want something more robust that a string. Otherwise we still face the issue of theme being badly written with "boxxed" and nobody will ever see it.

- Maybe start with `logo-height`, it seems to be the easiest: this a numerical value. You can use a char field, to make it contain the unit as well for now.
- For "layout", in a later commit:
   - We could use a Selection field... but this would mean it is impossible to extend for our community theme, bad.
   - This "layout" here should rather be about making a Many2one towards the model we talked about: `website.design.option`, only allowing records that have the "layout" variable name. Something like that. The view_id field of the option record would just be empty.
=> Doing that would have the same benefits we talked about: the editor displays all "layout" options automatically. No need of xpathing the right editor widget, defining the button the right way, etc to create a new layout type => just need to create a `website.design.option` record.